nomad/job_endpoint.go

```diff
@@ -328,52 +349,51 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 			return err
 		}
 
+		submittedEval = true
+
 		// Populate the reply with job information
 		reply.JobModifyIndex = index
+		reply.Index = index
+
+		if args.Eval != nil {
+			reply.EvalCreateIndex = index
+		}
+
 	} else {
 		reply.JobModifyIndex = existingJob.JobModifyIndex
 	}
 
 	// used for multiregion start
 	args.Job.JobModifyIndex = reply.JobModifyIndex
 
-	// If the job is periodic or parameterized, we don't create an eval.
-	if args.Job.IsPeriodic() || args.Job.IsParameterized() {
+	if args.Eval == nil {
 		return nil
 	}
 
-	// Create a new evaluation
-	now := time.Now().UTC().UnixNano()
-	eval := &structs.Evaluation{
-		ID:             uuid.Generate(),
-		Namespace:      args.RequestNamespace(),
-		Priority:       args.Job.Priority,
-		Type:           args.Job.Type,
-		TriggeredBy:    structs.EvalTriggerJobRegister,
-		JobID:          args.Job.ID,
-		JobModifyIndex: reply.JobModifyIndex,
-		Status:         structs.EvalStatusPending,
-		CreateTime:     now,
-		ModifyTime:     now,
-	}
-	update := &structs.EvalUpdateRequest{
-		Evals:        []*structs.Evaluation{eval},
-		WriteRequest: structs.WriteRequest{Region: args.Region},
-	}
+	// COMPAT(1.1.0): Remove the ServerMeetMinimumVersion check.
```

Comment:
I think this might be clearer if we said we want to remove the entire block this condition guards (and the `submittedEval` flag), and not just the check itself.

Reply:
I think we'll need to keep block with `submittedEval` check.  Once we remove compatibility, we'll still want to emit a EvalUpdateRequest log entry for when the RPC doesn't change job - in such case, no job registration request is emitted, and `submittedEval` is false.

Reply:
Oh, right! 👍